logstash-core/spec/api/lib/api/plugins_spec.rb

```diff
@@ -0,0 +1,45 @@
+# encoding: utf-8
+require_relative "../../spec_helper"
+require "sinatra"
+require "app/modules/plugins"
+require "logstash/json"
+
+describe LogStash::Api::Plugins do
+
+  include Rack::Test::Methods
+
+  def app()
+    described_class
+  end
+
+  before(:all) do
+    get "/"
+  end
+
+  let(:payload) { JSON.parse(last_response.body) }
+
+  it "respond to plugins resource" do
+    expect(last_response).to be_ok
+  end
+
+  it "return valid json content type" do
+    expect(last_response.content_type).to eq("application/json")
+  end
+
+  it "return the expected schema" do
+    expect(payload.keys).to include("plugins", "total")
+    payload["plugins"].each do |plugin|
+      expect(plugin.keys).to include("name", "version")
```

Comment:
Hi,
   do you want to also make sure we return plugin names with content and
following our format?

 should I extend this way of testing to each other test for the webapi?

  - purbon


On Mon, Mar 14, 2016 at 3:53 PM Pier-Hugues Pellerin <
notifications@github.com> wrote:

> In logstash-core/spec/api/lib/api/plugins_spec.rb
> <https://github.com/elastic/logstash/pull/4794#discussion_r56012544>:
>
> > +  end
> > +
> > +  let(:payload) { JSON.parse(last_response.body) }
> > +
> > +  it "respond to plugins resource" do
> > +    expect(last_response).to be_ok
> > +  end
> > +
> > +  it "return valid json content type" do
> > +    expect(last_response.content_type).to eq("application/json")
> > +  end
> > +
> > +  it "return the expected schema" do
> > +    expect(payload.keys).to include("plugins", "total")
> > +    payload["plugins"].each do |plugin|
> > +      expect(plugin.keys).to include("name", "version")
>
> Agree, it wont happen when you know the implementation, but when we
> refactor the implementation to share code with the plugin manager this
> statement will certainly changes.
>
> So IMHO, I think we shouldn't use include? unless we don't actually mind
> sending nil values
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/elastic/logstash/pull/4794/files#r56012544>.
>